docs/RELEASE.md

```diff
@@ -0,0 +1,77 @@
+# Release
+
+This document aims to describe how to perform a release of the Horreum python library.
+
+The versioning pattern should follow the [Horreum](https://github.com/Hyperfoil/Horreum) 
+versioning scheme to keep coherence among all Horreum-related projects versions.
+
+## Procedure
+
+### Tag a new version
+
+Checkout to your branch, either a "stable" (e.g., `0.12.x`) or the `main` one.
+
+```bash
+git checkout origin/main
+```
+
+Update the project version:
+
+```bash
+poetry version patch
+```
+
+This will bump your version, from `0.12-dev` to `0.12`.
+
+To double-check the version, run:
+```bash
+poetry version
+# horreum 0.12
+```
+
+Commit the changes and tag a new version:
+```bash
+git add .
+git commit -m "Tag version 0.12"
+git tag v0.12
+```
+
+Ensure the tag is in the form of `v$(poetry version)`.
+
+Push changes and tag:
+```bash
+git push origin main
+git push origin v0.12
+```
+
+If you are releasing a new patch from a _stable_ branch all previous operations must be performed
+from _stable_ rather than from `main`.
+
+### Create stable branch
+
+> **NOTE**: If the _stable_ branch already exists, simply skip this step, as this means the following steps have already been done.
+
+To create a _stable_ branch from the `main` one, e.g., `0.12.x`, run the following commands.
+
+```bash
+git checkout origin/main -b 0.12.x
+git checkout origin/main
+```
```

Comment:
Again, before checking out `origin/main` you need to make sure that it is up to date with the `origin` repo, either by doing a `git remote update` or a `git pull`.

And, when I check out the `main` branch, I need to specify the local branch name:
```
git checkout origin/main -b main
```
otherwise, I end up with a "detached HEAD", and the attempt to `push` my local `main` branch won't push the changes that `poetry` made (because they will be on my detached branch and not on `main`, if it even exists).